contracts/babylon/src/contract.rs

```diff
@@ -7,24 +7,24 @@ use cw_utils::ParseReplyError;
 
 use babylon_apis::{btc_staking_api, finality_api};
 use babylon_bindings::BabylonMsg;
+use btc_light_client::msg::contract::InstantiateMsg as BtcLightClientInstantiateMsg;
 
 use crate::error::ContractError;
 use crate::ibc::{ibc_packet, IBC_CHANNEL, IBC_TRANSFER};
 use crate::msg::contract::{ContractMsg, ExecuteMsg, InstantiateMsg, QueryMsg};
 use crate::queries;
-use crate::state::btc_light_client;
 use crate::state::config::{Config, CONFIG};
 
 pub const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
 pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-const REPLY_ID_INSTANTIATE_STAKING: u64 = 2;
-const REPLY_ID_INSTANTIATE_FINALITY: u64 = 3;
+const REPLY_ID_INSTANTIATE_LIGHT_CLIENT: u64 = 2;
+const REPLY_ID_INSTANTIATE_STAKING: u64 = 3;
+const REPLY_ID_INSTANTIATE_FINALITY: u64 = 4;
 
-/// When we instantiate the Babylon contract, it will optionally instantiate a BTC staking
-/// contract – if its code id is provided – to work with it for BTC re-staking support,
-/// as they both need references to each other.
-/// The admin of the BTC staking contract is taken as an explicit argument.
+/// When we instantiate the Babylon contract, it will optionally instantiate a BTC light client
+/// contract first – if its code id is provided – followed by BTC staking and finality contracts
+/// if their code ids are provided.
```

Comment:
Let's be careful with the sequence here. This will require modification to / handling in the instantiate msg in the SDK.

Reply:
Yes I deliberately instantiate BTC light client earlier than others. Added a dostring to make it explicit